src/bench/cluster_linearize.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+
+#include <util/bitset.h>
+#include <cluster_linearize.h>
+
+using namespace cluster_linearize;
+
+namespace {
+
+/** Construct a linear graph. These are pessimal for AncestorCandidateFinder, as they maximize
+ *  the number of ancestor set feerate updates. The best ancestor set is always the topmost
+ *  remaining transaction, whose removal requires updating all remaining transactions' ancestor
+ *  set feerates. */
+template<typename SetType>
+DepGraph<SetType> MakeLinearGraph(ClusterIndex ntx)
+{
+    DepGraph<SetType> depgraph;
+    for (ClusterIndex i = 0; i < ntx; ++i) {
+        depgraph.AddTransaction({-int32_t(i), 1});
+        if (i > 0) depgraph.AddDependency(i - 1, i);
+    }
+    return depgraph;
+}
+
+// Construct a difficult graph. These need at least sqrt(2^(n-1)) iterations in the best
+// known algorithms (purely empirically determined).
+template<typename SetType>
+DepGraph<SetType> MakeHardGraph(ClusterIndex ntx)
```

Comment:
Matches my diagrams. Thanks for the additional context! :+1:

Reply:
Ah I didn't notice this yesterday - I think you may have swapped odd and even? Same in the commented mermaid diagram code just added.

Reply:
Hm, I was confused because when I was looking at the code, I expected the branch with `depgraph.AddDependency(i, 4)` to have lots of T4-->Tx in its diagram. And the branch with `depgraph.AddDependency(i, 3)` to have lots of T3-->Tx.

{T0...T9} is 10 transactions, and {T0...T10} is 11 transactions. Am I missing something?

Reply:
Whew, so not wholly unmeritorious.